tdrs-backend/tdpservice/parsers/schema_defs/tanf/t7.py

```diff
@@ -0,0 +1,51 @@
+"""Schema for TANF T7 Row."""
+
+from ...util import SchemaManager
+from ...fields import Field, TransformField
+from ...row_schema import RowSchema
+from ...transforms import calendar_quarter_to_rpt_month_year
+from ... import validators
+from tdpservice.search_indexes.models.tanf import TANF_T7
+
+schemas = []
+
+validator_index = 7
+section_ind_index = 7
+stratum_index = 8
+families_index = 10
+for i in range(1, 31):
+    schemas.append(
+      RowSchema(
+          model=TANF_T7,
+          quiet_preparser_errors=i > 1,
+          preparsing_validators=[
+              validators.notEmpty(0, 7),
+              validators.notEmpty(validator_index, validator_index + 24),
+          ],
+          postparsing_validators=[],
+          fields=[
+              Field(item="0", name="RecordType", type='string', startIndex=0, endIndex=2,
+                    required=True, validators=[]),
+              Field(item="3", name='CALENDAR_QUARTER', type='number', startIndex=2, endIndex=7,
+                    required=True, validators=[validators.dateYearIsLargerThan(1998),
+                                               validators.quarterIsValid()]),
+              TransformField(calendar_quarter_to_rpt_month_year(i % 3), item="3A", name='RPT_MONTH_YEAR', type='number',
```

Comment:
@jtimpe which data element does 3A map to?

I'm seeing unexpected results. in the screenshots below, 1, 2 and 3 represent months 1, 2, and 3. 

_this is how i expected the data to look after restructuring to one record per tdrs indicator and stratum_
![1111p1](https://github.com/raft-tech/TANF-app/assets/63075587/c9111351-2cb8-4a16-8f16-1b40e3e87cb8)

_this is how i expected the data to look after restructuring to one record per tdrs indicator, stratum and month_
![1111p2](https://github.com/raft-tech/TANF-app/assets/63075587/b64be748-a1b9-44bf-aa4b-8ed54a950d12)

_this is what im seeing in DAC_
![1111p3](https://github.com/raft-tech/TANF-app/assets/63075587/df334d47-2fc0-409f-a8b5-54d3d5daae91)

Reply:
good catch. we were creating schemas starting with index `1` but assumed the first record would have index `0` for the month transform. offsetting the transform value appears to fix it, and i've updated the test cases to reflect that.